javascript-data-model/main.js

```diff
@@ -0,0 +1,45 @@
+/* exported orderHistory */
+
+var orderHistory = [
+  {
+    name: "JavaScript for impatient programmers",
+    type: "book",
+    by: "Rauschmayer, Dr. Axel",
+    price: 31.55,
+    deliveredDate: 08082020,
+    returnByDate: 09072020
+  },
+  {
+    name: "The Timeless Ways of Building",
+    type: "book",
+    by: "Alexander, Christopher",
+    price: 41.33,
+    deliveredDate: 07202020,
+    returnByDate: 0819202
+  },
+  {
+    name: "Gamecube Controller Adapter",
+    type: ["wii U", "PC"],
```

Comment:
But here, `type` is an array of strings. When the data type of the value differs like this between entries, looping over your data becomes a nightmare. Try to make sure that your data model and the types each value holds are the same between all entries. The data itself will vary, but the structure they are represented in should always be the same.